tests/integration/test_wazuh_db/test_db_backup.py

```diff
@@ -202,17 +202,20 @@ def test_wdb_backup_command(configure_sockets_environment, connect_to_sockets_mo
         assert test_values[0] in db_response[-1]['key'], f'Expected value key:"{test_values[0]}" was not found.'
 
         if save_pre_restore == 'true':
-            backups= query_wdb(get_backups_command)
+            backups = query_wdb(get_backups_command)
             # Check that the pre-restore state backup has been generated.
-            assert backups.__len__() ==  backups_amount +1, f'Found {backups.__len__()} files, \
+            assert backups.__len__() == backups_amount + 1, f'Found {backups.__len__()} files, \
                                                                expected {backups_amount + 1}'
-            assert "-pre_restore.gz" in backups[-1], f'Did not find the expected "-pre_restore.gz" file"'
+            # Combine the backups list content because in 'Ubuntu' the value are not stored in order.
+            combined = '\t'.join(backups)
+            assert "-pre_restore.gz" in combined, f'Did not find the expected "-pre_restore.gz" file"'
 
             if 'restore_pre_restore' in case_data:
                 restore_command = f'global backup restore {{"snapshot": "{backups[-1]}",\
```

Comment:
You are always restoring the last element, but if you don't know the order, you could restore the wrong file. Please, check this

Reply:
Done [here](https://github.com/wazuh/wazuh-qa/pull/3802/commits/6447acb68878bb893ad57c41b065e1bfd953a017)